File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people:list, limit:int):
    answer = 0
    people.sort()

    s = 0
    e = len(people) - 1
    while s < e:
        if limit >= people[s] + people[e]:
            s += 1
            e -= 1
            answer += 1
        else:
            e -= 1
            answer += 1
        
    if s == e:
        answer += 1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);

        int answer = 0;
        for (int s = 0, e = people.length - 1; s < e; e--) {
            if (limit >= people[s] + people[e]) {
                s++;
                answer++;
            }
        }

        return people.length - answer;
    }
}